File: day4/day4.js
const fs = require('fs');
const CryptoJS = require('crypto-js');
const input = fs.readFileSync("day4/day4.txt", 'utf-8');

function part1() {
    var i = 1;
    while (!CryptoJS.MD5(input + i).toString().startsWith("00000")) i++;
    return i;
}

function part2() {
    var i = 1;
    while (!CryptoJS.MD5(input + i).toString().startsWith("000000")) i++;
    return i;
}

console.log("Part 1: " + part1());
console.log("Part 2: " + part2());

File: day4/day4.py
import hashlib

data = open('day4/day4.txt').read()

def part1():
    i = 1
    while not hashlib.md5(f"{data}{i}".encode()).hexdigest().startswith('00000'):
        i += 1
    return i

def part2():
    i = 1
    while not hashlib.md5(f"{data}{i}".encode()).hexdigest().startswith('000000'):
        i += 1
    return i

print(f'Part 1: {part1()}')
print(f'Part 2: {part2()}')